config_models/models.py

```diff
@@ -65,18 +58,17 @@ def with_active_flag(self):
         if it's the most recent entry for that combination of keys.
         """
         if self.model.KEY_FIELDS:
-            subquery = self._current_ids_subquery()
-            return self.get_queryset().extra(           # pylint: disable=no-member
-                select={'is_active': "{table_name}.id IN ({subquery})".format(
-                    table_name=self.model._meta.db_table,  # pylint: disable=protected-access, no-member
-                    subquery=subquery,
-                )}
+            return self.get_queryset().annotate(
+                is_active=models.ExpressionWrapper(
+                    models.Q(pk__in=self._current_ids_subquery()),
+                    output_field=models.IntegerField(),
+                )
+            )
+        return self.get_queryset().annotate(
+            is_active=models.ExpressionWrapper(
+                models.Q(pk=self.model.current().pk),
+                output_field=models.IntegerField(),
```

Comment:
Historical question: Was the previous method for doing these custom model manager queries done with raw SQL and different methods because the methods which this PR uses didn't yet exist when this code was written? Is this change just an evolution to use the more appropriate methods for the purpose?

Reply:
I think annotate existed at the time the old code was written. But I suspect that the original author (maybe me?) didn't know about them.